components/tools/OmeroWeb/test/integration/test_api_images.py

```diff
@@ -0,0 +1,182 @@
+        """Return Dataset with Images and an orphaned Image."""
+        query = get_query_service(user1)
+        dataset = DatasetI()
+        dataset.name = rstring('Dataset')
+
+        # Create 5 Images in Dataset
+        for i in range(5):
+            img = self.create_test_image(size_x=125, size_y=125,
+                                         session=user1[0].getSession(),
+                                         name="Image%s" % i)
+            img = ImageI(img.id.val, False)
+            dataset.linkImage(img)
+
+        # Import a single orphaned Image "tinyTest.d3d.dv" and get ImageID
+        pids = self.import_image(client=user1[0], skip=None)
+        pixels = query.get("Pixels", long(pids[0]))
+        image = query.get("Image", pixels.image.id.val)
+
+        dataset = get_update_service(user1).saveAndReturnObject(dataset)
+        return dataset, image
+
+    def test_dataset_images(self, user1, dataset_images):
+        """Test listing of Images in a Dataset."""
+        conn = get_connection(user1)
+        user_name = conn.getUser().getName()
+        django_client = self.new_django_client(user_name, user_name)
+        version = settings.API_VERSIONS[-1]
+
+        dataset = dataset_images[0]
+        images = dataset.linkedImageList()
+        orphaned = dataset_images[1]
+
+        images_url = reverse('api_images', kwargs={'api_version': version})
+        datasets_url = reverse('api_datasets', kwargs={'api_version': version})
+
+        # List ALL Images
+        rsp = _get_response_json(django_client, images_url, {})
+        assert len(rsp['data']) == 6
+
+        # Filter Images by Orphaned
+        payload = {'orphaned': 'true'}
+        rsp = _get_response_json(django_client, images_url, payload)
+        assert_objects(conn, rsp['data'], [orphaned], dtype='Image',
+                       opts={'load_pixels': True})
+
+        # Filter Images by Dataset
+        images.sort(cmp_name_insensitive)
+        payload = {'dataset': dataset.id.val}
+        rsp = _get_response_json(django_client, images_url, payload)
+        # Manual check that Pixels is loaded but Channels are not
+        assert 'Channels' not in rsp['data'][0]['Pixels']
+        assert_objects(conn, rsp['data'], images, dtype='Image',
+                       opts={'load_pixels': True})
+
+        # Pagination, listing images via /datasets/:id/images/
+        limit = 3
+        dataset_images_url = datasets_url + "%s/images/" % dataset.id.val
+        payload = {'dataset': dataset.id.val, 'limit': limit}
+        rsp = _get_response_json(django_client, dataset_images_url, payload)
+        assert_objects(conn, rsp['data'], images[0:limit], dtype='Image',
+                       opts={'load_pixels': True})
+        payload['page'] = 2
+        rsp = _get_response_json(django_client, images_url, payload)
+        assert_objects(conn, rsp['data'], images[limit:limit * 2],
+                       dtype='Image', opts={'load_pixels': True})
```

Comment:
In this case I'm not testing the loading of images with/without loading pixels. We don't provide that option for web api (always load pixels). The ```opts={'load_pixels': True}``` is just required here so that the when we ```getObjects('Image', opts={'load_pixels': True})``` to compare objects to json, we are comparing like-for-like (pixels loaded in both cases).